tools/kontaktdaten.py

```diff
@@ -29,6 +32,76 @@ def get_kontaktdaten(filepath):
         return {}
 
 
+def check_kontaktdaten(kontaktdaten: dict, mode: Modus):
+    """
+    Überprüft ob alle Keys vorhanden sind und ob die Values kein leeren String enthalten
+
+    Args:
+        mode (Modus): Entsprechend werden Daten überprüft
+        kontaktdaten (dict): Inhalt der JSON
+
+    Raises:
+        MissingValuesError: Es wird ein Key oder Value vermisst
+    """
+
+    if mode == Modus.TERMIN_SUCHEN:
+        try:
+            # Daten vorhanden
+            kontaktdaten["plz_impfzentren"]
+            kontaktdaten["code"]
+            kontaktdaten["kontakt"]["anrede"]
+            kontaktdaten["kontakt"]["vorname"]
+            kontaktdaten["kontakt"]["nachname"]
+            kontaktdaten["kontakt"]["strasse"]
+            kontaktdaten["kontakt"]["hausnummer"]
+            kontaktdaten["kontakt"]["plz"]
+            kontaktdaten["kontakt"]["ort"]
+            kontaktdaten["kontakt"]["phone"]
+            kontaktdaten["kontakt"]["notificationChannel"]
+            kontaktdaten["kontakt"]["notificationReceiver"]
+
+            # Daten enthalten kein leerer String
+            # PLZ
+            for plz in kontaktdaten["plz_impfzentren"]:
+                if not plz.strip():
+                    raise MissingValuesError("Wert fuer \"plz_impfzentren\" fehlerhaft!")
+            if not kontaktdaten["code"].strip():
+                raise MissingValuesError("Wert fuer \"code\" fehlt!")
+
+            # 2. Ebene von Kontakt
+            for key, value in kontaktdaten["kontakt"].items():
+                if not value.strip():
+                    raise MissingValuesError(f"Wert fuer \"{key}\" fehlt!")
```

Comment:
Die letzten 12 Zeilen dürften unnötig sein, wenn man auch validiert.

Reply:
ich weiß jetzt nicht genau welche 12 Zeilen / 14 Zeilen du genau meinst.

Wenn es um die Überprüfung der leeren Strings geht, dann stimmt das teilweise.
Die PLZ_Impfzentren, Code, Handynummer und Mail könnte man theoretisch weglassen, da hier später in der Validierung nicht nur auf den Typ geprüft wird, sondern Daten vorhanden sein müssen.

Die Validierung von anrede, vorname, nachname, strasse, hausnummer, ort und notificationChannel überprüft nicht ob Daten vorhanden sind, sondern nur ob es ein String ist. Heißt leere Strings wären valide.

Das doppelt gemoppelt habe ich aus 2 Gründen drin gelassen.
- Die Methode macht wirklich genau das was sie tun soll / wie sie beschrieben ist - Prüfen ob Daten fehlen. Unabhängig ob man diese dann später nochmals validiert.
- Leebarkeit des Code - Nicht das sich die Frage gestellt wird "Warum prüft man jetzt nicht alle Daten, obwohl die ja auch leer sein könnten?"

Reply:
Ok, aber wenn wir leere Strings verbieten wollen, dann wäre es ja viel sinnvoller, diese in der Validation zu verbieten, damit man sie erst gar nicht über die GUI eingeben kann.